carla/evaluation/process_nans.py

```diff
@@ -0,0 +1,28 @@
+from typing import Tuple
+
+import pandas as pd
+
+
+def remove_nans(
+    factuals: pd.DataFrame, counterfactuals: pd.DataFrame
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """
+
+    Parameters
+    ----------
+    factuals: Has to be the same shape as counterfactuals
+    counterfactuals: Has to be the same shape as factuals
+
+    Returns
+    -------
+
+    """
+    nan_idx = counterfactuals.index[counterfactuals.isnull().any(axis=1)]
+
+    output_factuals = factuals.copy()
+    output_counterfactuals = counterfactuals.copy()
+
+    output_factuals = output_factuals.drop(index=nan_idx)
```

Comment:
Alright, makes sense. Was just a bit worried that this can lead to incorrect results. If I remember correctly the number of counterfactuals per factual was a parameter that was changable